### PR TITLE
Sort classifiers and respect `--indent`

### PR DESCRIPTION
**Pull request by hugovk**
Right now, pyproject-fmt indents most sections using the `--indent` value (default: 2 spaces), but not the project `classifiers` section.

For example, if I first modify this project's `pyproject.toml` to use four space, then run `pyproject-fmt pyproject.toml`, they're all fixed to two spaces, except for the classifiers:

```diff
--- pyproject.toml

+++ pyproject.toml

@@ -1,8 +1,8 @@

 [build-system]
 build-backend = "hatchling.build"
 requires = [
-    "hatch-vcs>=0.3",
-    "hatchling>=1.13",
+  "hatch-vcs>=0.3",
+  "hatchling>=1.13",
 ]

 [project]
@@ -10,8 +10,8 @@

 description = "Format your pyproject.toml file"
 readme = "README.md"
 keywords = [
-    "format",
-    "pyproject",
+  "format",
+  "pyproject",
 ]
 license.file = "LICENSE.txt"
 authors = [{ name = "Bernat Gabor", email = "gaborjbernat@gmail.com" }]
@@ -28,25 +28,25 @@

     "Programming Language :: Python :: 3.11",
 ]
 dynamic = [
-    "version",
+  "version",
 ]
 dependencies = [
-    "packaging>=23",
-    "tomlkit>=0.11.6",
-    'typing-extensions>=4.5; python_version < "3.8"',
+  "packaging>=23",
+  "tomlkit>=0.11.6",
+  'typing-extensions>=4.5; python_version < "3.8"',
 ]
 optional-dependencies.docs = [
-    "furo>=2022.12.7",
-    "sphinx>=6.1.3",
-    "sphinx-argparse-cli>=1.11",
-    "sphinx-autodoc-typehints!=1.23.4,>=1.22",
-    "sphinx-copybutton>=0.5.1",
+  "furo>=2022.12.7",
+  "sphinx>=6.1.3",
+  "sphinx-argparse-cli>=1.11",
+  "sphinx-autodoc-typehints!=1.23.4,>=1.22",
+  "sphinx-copybutton>=0.5.1",
 ]
 optional-dependencies.test = [
-    "covdefaults>=2.3",
-    "pytest>=7.2.2",
-    "pytest-cov>=4",
-    "pytest-mock>=3.10",
+  "covdefaults>=2.3",
+  "pytest>=7.2.2",
+  "pytest-cov>=4",
+  "pytest-mock>=3.10",
 ]
 urls."Bug Tracker" = "https://github.com/tox-dev/pyproject-fmt/issues"
 urls."Changelog" = "https://github.com/tox-dev/pyproject-fmt/releases"
```

With this PR, the classifiers are also sorted and indented:

```diff
--- pyproject.toml

+++ pyproject.toml

@@ -1,8 +1,8 @@

 [build-system]
 build-backend = "hatchling.build"
 requires = [
-    "hatch-vcs>=0.3",
-    "hatchling>=1.13",
+  "hatch-vcs>=0.3",
+  "hatchling>=1.13",
 ]

 [project]
@@ -10,44 +10,44 @@

 description = "Format your pyproject.toml file"
 readme = "README.md"
 keywords = [
-    "format",
-    "pyproject",
+  "format",
+  "pyproject",
 ]
 license.file = "LICENSE.txt"
 authors = [{ name = "Bernat Gabor", email = "gaborjbernat@gmail.com" }]
 requires-python = ">=3.7"
 classifiers = [
-    "License :: OSI Approved :: MIT License",
-    "Operating System :: OS Independent",
-    "Programming Language :: Python",
-    "Programming Language :: Python :: 3 :: Only",
-    "Programming Language :: Python :: 3.7",
-    "Programming Language :: Python :: 3.8",
-    "Programming Language :: Python :: 3.9",
-    "Programming Language :: Python :: 3.10",
-    "Programming Language :: Python :: 3.11",
+  "License :: OSI Approved :: MIT License",
+  "Operating System :: OS Independent",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.7",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
 ]
 dynamic = [
-    "version",
+  "version",
 ]
 dependencies = [
-    "natsort",
-    "packaging>=23",
-    "tomlkit>=0.11.6",
-    'typing-extensions>=4.5; python_version < "3.8"',
+  "natsort",
+  "packaging>=23",
+  "tomlkit>=0.11.6",
+  'typing-extensions>=4.5; python_version < "3.8"',
 ]
 optional-dependencies.docs = [
-    "furo>=2022.12.7",
-    "sphinx>=6.1.3",
-    "sphinx-argparse-cli>=1.11",
-    "sphinx-autodoc-typehints!=1.23.4,>=1.22",
-    "sphinx-copybutton>=0.5.1",
+  "furo>=2022.12.7",
+  "sphinx>=6.1.3",
+  "sphinx-argparse-cli>=1.11",
+  "sphinx-autodoc-typehints!=1.23.4,>=1.22",
+  "sphinx-copybutton>=0.5.1",
 ]
 optional-dependencies.test = [
-    "covdefaults>=2.3",
-    "pytest>=7.2.2",
-    "pytest-cov>=4",
-    "pytest-mock>=3.10",
+  "covdefaults>=2.3",
+  "pytest>=7.2.2",
+  "pytest-cov>=4",
+  "pytest-mock>=3.10",
 ]
 urls."Bug Tracker" = "https://github.com/tox-dev/pyproject-fmt/issues"
 urls."Changelog" = "https://github.com/tox-dev/pyproject-fmt/releases"
```

Note we need to use something like https://github.com/SethMMorton/natsort for the classifiers, because they can include version numbers, and the basic `sorted` gives:

```diff
   "Operating System :: OS Independent",
   "Programming Language :: Python",
   "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
   "Programming Language :: Python :: 3.7",
   "Programming Language :: Python :: 3.8",
   "Programming Language :: Python :: 3.9",
-  "Programming Language :: Python :: 3.10",
-  "Programming Language :: Python :: 3.11",
 ]
```

### CODE REVIEW

**Comment by hugovk** [comment]
mypy error:
```
src/pyproject_fmt/formatter/util.py:102: error: Cannot call function of unknown type  [operator]
Found 1 error in 1 file (checked 11 source files)
```

Line 102 is the second one here:
```python
    sort_method = natsorted if custom_sort == "natsort" else sorted
    for element in sort_method(entries, key=key):
```

What annotation do we need here?

File: src/pyproject_fmt/formatter/util.py [hunk 1]
```diff
@@ -94,7 +98,8 @@ def sorted_array(
     indent_text = " " * indent
     for start_entry in start:
         body.append(_ArrayItemGroup(indent=Whitespace(f"\n{indent_text}"), comment=start_entry))
-    for element in sorted(entries, key=key):
+    sort_method = natsorted if custom_sort == "natsort" else sorted
```
Comment (gaborbernat) [inline src/pyproject_fmt/formatter/util.py:101]: natsorted -> not_sorted

Comment (hugovk) [reply src/pyproject_fmt/formatter/util.py:101]: Like this?

```python
from natsort import natsorted as nat_sorted

...

    sort_method = nat_sorted if custom_sort == "natsort" else sorted
```

File: src/pyproject_fmt/formatter/util.py [hunk 2]
```diff
@@ -75,7 +76,10 @@ class ArrayEntries:
 
 
 def sorted_array(
-    array: Array | None, indent: int, key: Callable[[ArrayEntries], str] = lambda e: str(e.text).lower()
+    array: Array | None,
+    indent: int,
+    key: Callable[[ArrayEntries], str] = lambda e: str(e.text).lower(),
+    custom_sort: str | None = None,
```
[the remark this answers is not in the document]
Comment (hugovk) [reply src/pyproject_fmt/formatter/util.py:82]: How do you mean?

**Review by gaborbernat — CHANGES_REQUESTED** [review]
Can you fix the type checker and resolve the merge conflict please?

**Comment by hugovk** [comment]
Merge conflict resolved, I'm not sure about the type checker but will ask someone here at PyCon :)

**Comment by hugovk** [comment]
Typing fixed with some help :)

image

**Comment by gaborbernat** [comment]
I'm also at the conference, should meet up tomorrow 😂

**Pull request merged**